otter/test/convergence/test_steps.py

```diff
@@ -451,21 +453,28 @@ def test_change_load_balancer_node(self):
         :obj:`ChangeCLBNode.as_effect` produces a request for
         modifying a load balancer node.
         """
-        changenode = ChangeCLBNode(
+        change_node = ChangeCLBNode(
             lb_id='abc123',
             node_id='node1',
             condition=CLBNodeCondition.DRAINING,
             weight=50,
             type=CLBNodeType.PRIMARY)
-        self.assertEqual(
-            changenode.as_effect(),
-            service_request(
-                ServiceType.CLOUD_LOAD_BALANCERS,
-                'PUT',
-                'loadbalancers/abc123/nodes/node1',
-                data={'condition': 'DRAINING',
-                      'weight': 50},
-                success_pred=has_code(202)))
+        eff = change_node.as_effect()
+
+        handled_codes = _clb_check_change_node_handlers.keys()
+        expected_intent = service_request(
+            ServiceType.CLOUD_LOAD_BALANCERS,
+            'PUT',
+            'loadbalancers/abc123/nodes/node1',
+            data={'condition': 'DRAINING',
+                  'weight': 50},
+            success_pred=has_code(*handled_codes)).intent
+        self.assertEqual(eff.intent, expected_intent)
+
+        (on_success, on_error), = eff.callbacks
+        self.assertIdentical(on_error, None)
+        self.assertIdentical(on_success.func, _clb_check_change_node)
+        self.assertEqual(on_success.args, (change_node,))
```

Comment:
Non-blocking: This is literal testing of callback functions. Would you consider testing by resolving effect with status codes and looking at the result? I can understand if you don't want to do that since there are separate tests for `_clb_check_change_node` and it is expected to be used elsewhere.